datashape/discovery.py

```diff
@@ -362,9 +362,10 @@ def unpack(ds):
         return ds
 
 
-@dispatch(dict)
-def discover(d):
-    return Record([[k, discover(d[k])] for k in sorted(d)])
+@discover.register(dict)
+@discover.register(MappingProxyType)
+def _mapping_discover(m):
+    return Record((k, discover(m[k])) for k in sorted(m))
```

Comment:
I could add another dispatch for ordereddict which preserves order. I think we want to sort mappings in general because two dicts that are the same kv types should give the same dshape but ordereddict seems special here. For user-defined ordered types they would just need to override discover themselves.